### DISCUSSION TITLE
User model - profile callback returns only default fields

### DISCUSSION THREAD

**Discussion by AdisonCavani** [nextauthjs/next-auth#6731] [Help] [answered]
next-auth 4.19.2

I've setup `next-auth` with AWS Cognito Provider.
I want to retrieve additional fields, so I customized user model via `profile` callback.
Docs: https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider

```ts
providers: [
    CognitoProvider({
      clientId: process.env.COGNITO_CLIENT_ID,
      clientSecret: process.env.COGNITO_CLIENT_SECRET,
      issuer: process.env.COGNITO_ISSUER,

      profile(profile) {
        return {
          id: "123",
          email: "test@email.com",
          image: "jpg",
          address: "myAddress",
        };
      },
    }),
  ],
```

```tsx
const session = useSession();
console.log(session.data?.user);
```

Result:
```json
{
    "email": "adrian@email.com",
    "image": "jpg"
}
```

It should return `id, email, image, address`. 
How to fix this issue?

**Comment by adarah** [comment, marked as answer]
You will receive the profile value in the [jwt callback](https://next-auth.js.org/configuration/callbacks#jwt-callback). From there you need to attach it to the token. Then, in the session callback, grab the value from the token and assign it to the session.

**Reply by AdisonCavani** [reply]
Thank you.
